src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -377,7 +536,25 @@ private JButton getNewButton() {
 							MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
 							medicalInventoryRow.setInventory(meInventory);
 							Lot lot = medicalInventoryRow.getLot();
-							if (lot != null && lot.getCode().equals("")) {
+							String lotCode;
+							Medical medical = medicalInventoryRow.getMedical();
+							if (lot != null) {
+								lotCode = lot.getCode();
+								Lot lotExist = movStockInsertingManager.getLot(lotCode);
+								if (lotExist != null) {
+									Lot lotStore = null;
+									lotStore = movStockInsertingManager.updateLot(lot);
+									medicalInventoryRow.setLot(lotStore);
+								} else {
+									if (lot.getDueDate() != null) {
+										Lot lotStore = movStockInsertingManager.storeLot(lotCode, lot, medical);
+										medicalInventoryRow.setLot(lotStore);
+										medicalInventoryRow.setNewLot(true);
+									} else {
+										medicalInventoryRow.setLot(null);
+									}
+								}
+							} else {
 								medicalInventoryRow.setLot(null);
 							}
 							currentInventoryRow = medicalInventoryRowManager.newMedicalInventoryRow(medicalInventoryRow);
```

Comment:
in which case `currentInventoryRow` could be null? if any error occurs, it will be catched, or am I wrong?

Reply:
it will be catched on catch block

Reply:
exactly, so the following code should be useless, shouldn't it?

```
							if (currentInventoryRow == null) {
								checkResults++;
							}
```

Reply:
yes you right I will remove it